crypto/keys/ethsecp256k1/ethsecp256k1.go

```diff
@@ -0,0 +1,202 @@
+// where the last byte contains the recovery ID.
+func (privKey PrivKey) Sign(msg []byte) ([]byte, error) {
+	return ethcrypto.Sign(ethcrypto.Keccak256Hash(msg).Bytes(), privKey.ToECDSA())
+}
+
+// ToECDSA returns the ECDSA private key as a reference to ecdsa.PrivateKey type.
+// The function will panic if the private key is invalid.
+func (privKey PrivKey) ToECDSA() *ecdsa.PrivateKey {
+	key, err := ethcrypto.ToECDSA(privKey.Bytes())
+	if err != nil {
+		panic(err)
+	}
+	return key
+}
+
+// ----------------------------------------------------------------------------
+// secp256k1 Public Key
+
+var (
+	_ cryptotypes.PubKey   = &PubKey{}
+	_ codec.AminoMarshaler = &PubKey{}
+)
+
+// Address returns the address of the ECDSA public key.
+// The function will panic if the public key is invalid.
+func (pubKey PubKey) Address() tmcrypto.Address {
+	pubk, err := ethcrypto.DecompressPubkey(pubKey.Key)
+	if err != nil {
+		panic(err)
+	}
+
+	return tmcrypto.Address(ethcrypto.PubkeyToAddress(*pubk).Bytes())
+}
+
+// Bytes returns the raw bytes of the ECDSA public key.
+func (pubKey PubKey) Bytes() []byte {
+	return pubKey.Key
+}
+
+// String implements the fmt.Stringer interface.
+func (pubKey PubKey) String() string {
+	return fmt.Sprintf("EthPubKeySecp256k1{%X}", pubKey.Key)
+}
+
+// Type returns ethsecp256k1
+func (pubKey PubKey) Type() string {
+	return KeyType
+}
+
+// Equals returns true if the pubkey type is the same and their bytes are deeply equal.
+func (pubKey PubKey) Equals(other cryptotypes.PubKey) bool {
+	return pubKey.Type() == other.Type() && bytes.Equal(pubKey.Bytes(), other.Bytes())
+}
+
+// MarshalAmino overrides Amino binary marshalling.
+func (pubKey PubKey) MarshalAmino() ([]byte, error) {
+	return pubKey.Key, nil
+}
+
+// UnmarshalAmino overrides Amino binary marshalling.
+func (pubKey *PubKey) UnmarshalAmino(bz []byte) error {
+	if len(bz) != PubKeySize {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidPubKey, "invalid pubkey size, expected %d, got %d", PubKeySize, len(bz))
+	}
+	pubKey.Key = bz
+
+	return nil
+}
+
+// MarshalAminoJSON overrides Amino JSON marshalling.
+func (pubKey PubKey) MarshalAminoJSON() ([]byte, error) {
+	// When we marshal to Amino JSON, we don't marshal the "key" field itself,
+	// just its contents (i.e. the key bytes).
+	return pubKey.MarshalAmino()
+}
+
+// UnmarshalAminoJSON overrides Amino JSON marshalling.
+func (pubKey *PubKey) UnmarshalAminoJSON(bz []byte) error {
+	return pubKey.UnmarshalAmino(bz)
+}
+
+// VerifySignature verifies that the ECDSA public key created a given signature over
+// the provided message. It will calculate the Keccak256 hash of the message
+// prior to verification.
+//
+// CONTRACT: The signature should be in [R || S] format.
+func (pubKey PubKey) VerifySignature(msg []byte, sig []byte) bool {
+	if len(sig) == ethcrypto.SignatureLength {
+		// remove recovery ID (V) if contained in the signature
+		sig = sig[:len(sig)-1]
```

Comment:
It looks like this is forcing us to always use ECRecover? If so, can this be documented here, and a one line explanation added for `ethcrypto.SignatureLength`